### DISCUSSION TITLE
Does Class keyword work in Table, but not in TableRowCell?

### DISCUSSION THREAD

**Discussion by ghost** [Megabit/Blazorise#3393] [Q&A]
Ok, I'm a newbie to Web programming and to Blazorise, but...
I have a problem applying a css class to a Blazorise TableRowCell.

Here's the css class:
`
.test4b {
    background-color: green;
}
`

For Table, this does work: 
`<Table Borderless="true" Narrow="true" Style="table-layout:fixed;" Class="test4b">`
The background becomes green.

But for TableRowCell, this doesn't work:
`<TableRowCell Class="test4b">&nbsp;</TableRowCell>`
There is no effect, the background is the same as in the table (of course I tried this with a non-green table background ;-)

If I apply a style to the TableRowCell, this does work:
`<TableRowCell Style="background-color:lightgreen;">&nbsp;</TableRowCell>`

Am I doing something wrong or can't classes be applied to TableRowCell?

**Comment by stsrki** [comment]
Can you inspect it in DevTools and see what gets rendered in the HTML?

**Reply by ghost** [reply]
Sure. 

`<TableRowCell Style="background-color:lightgreen;">&nbsp;</TableRowCell>` renders to
![grafik](https://user-images.githubusercontent.com/10619439/152371056-0fbe26d0-9712-42d0-ab64-f5a76fb50391.png)

`<Table Borderless="true" Narrow="true" Style="table-layout:fixed;" Class="test4">` renders to
![grafik](https://user-images.githubusercontent.com/10619439/152371268-f5a253cd-c482-40fa-8c61-7802debfb7ff.png)

`<Table Borderless="true" Narrow="true" Style="table-layout:fixed;" Class="test4b">` renders to
![grafik](https://user-images.githubusercontent.com/10619439/152371418-2795329f-16eb-44dc-b455-8b1635775cb9.png)

I don't have a clue what this `_bl_952ba08b-d06d-4ac7-8702-21565dc87e98=""` means, but the background-color of the table is set to green, like mentioned in the test4b class.

**Comment by stsrki** [comment]
`_bl_952ba08b-d06d-4ac7-8702-21565dc87e98=""` is added by the Blazor when `@ref` is applied on the element. That part is OK.

You have 

![image](https://user-images.githubusercontent.com/900302/152373607-40760e35-a9c0-4eac-8801-f89c156b987b.png)

but what about the styles in DevTools? Can you inspect those? Just client on the element and it will show in the side area.

**Reply by ghost** [reply]
The background-color in the table is set to 0, 128, 0, which is correct for green.

For the lightgreen table cell (set via style), background-color is set to 144, 238, 144, which is correct, too.

In the tablecell which' class is set to test4, background-color is set to rgba(0, 0, 0, 0), which is black with no opacity, which is transparent.
But test4 is defined as
`.test4 {
    background-color: blue;
}
`, so the background-color should be rgb(0, 0, 255).

**Comment by ghost** [comment]
Update, I made some screenshots.

The table hass test4b as class:
![grafik](https://user-images.githubusercontent.com/10619439/152404024-4178fe30-0efb-413b-98c4-9615ff6bcc3b.png)

The table cell styled with lightgreen has no class:
![grafik](https://user-images.githubusercontent.com/10619439/152404070-e77b0cf5-00b1-4246-9674-bd3dc5792168.png)

And the table cell with class test4 has the class, but the style is strikethrough (see screenshot):
![grafik](https://user-images.githubusercontent.com/10619439/152404122-4c9ccbd4-5366-41c1-8624-65297e6fc890.png)
